mocks/helpers/buildDocFromHash.js

```diff
@@ -10,5 +10,15 @@ module.exports = function buildDocFromHash(hash = {}) {
       delete copy._ref;
       return copy;
     },
+    get(fieldPath) {
+      // The field path can be compound: from the firestore docs
+      //  fieldPath The path (e.g. 'foo' or 'foo.bar') to a specific field.
+      const parts = fieldPath.split('.');
+      let data = this.data();
+      parts.forEach(part => {
+        data = data[part];
+      });
+      return data;
```

Comment:
We iterate over all of the `parts` here to dig down deeper into `data`, for a varying number of keys.  The problem with this, however, is that `get()` in firestore is a safe method and will return `null` (I believe) if it can't find an example.

Say your `data` here looks like this:
```
{
  person: {
    name: 'bob'
  }
}
```
Your `get('person.name')` works, but `get('person.job.title')` would thrown an error since the second key would be undefined.  So, what if instead, we used a `reduce` method to safely drill down into that object:

```
get(fieldPath) {
  const parts = fieldPath.split('.');
  const data = this.data();

  return parts.reduce((acc, part, index) => {
   const value = acc[part];
   // if no key is found
   if (value === undefined) {
     // return null if we are on the last item in parts
     // otherwise, return an empty object, so we can continue to iterate
    return parts.length - 1 === index ? null : {};
  }

  // if there is a value, return it
  return value;
  }, data)
}
```

This allows us to safely drill through an object that may or may not have the keys we are looking for.  Do you think we can add a test case for when we can `record.get('foo.bar.baz')` but `bar` isn't actually defined?

Reply:
just an update, I've updated with your suggested fix, and as soon as the current batch of fires is out will update this PR w/ tests